src/player.cpp

```diff
@@ -11426,6 +11426,11 @@ void player::add_known_trap( const tripoint &pos, const trap &t)
     }
 }
 
+bool player::is_sleeping() const
+{
+    return has_effect( effect_sleep );
```

Comment:
No need to bloat the player class just for a tiny UI edge case. It is absolutely huge and full of functions used in 1-2 places each.
Either check `has_effect` directly or use sleep state. Skipping thunder messages for people who aren't yet fully asleep is a very minor effect.